src/test/java/test/guice/issue2343/modules/ParentModule.java

```diff
@@ -0,0 +1,27 @@
+package test.guice.issue2343.modules;
+
+import com.google.inject.AbstractModule;
+import java.util.Objects;
+import javax.inject.Singleton;
+import test.guice.issue2343.Person;
+
+public class ParentModule extends AbstractModule {
+
+  @Override
+  protected void configure() {
+    bind(Person.class).in(Singleton.class);
+  }
+
+  @Override
+  public boolean equals(final Object object) {
```

Comment:
Does it work without equals/hashCode overriding? I've never seen anyone overring these ones for guice modules.

Reply:
@juherr - To be honest, I haven't tried. I just borrowed the tests from the defect. I will try it out and update.

Reply:
@juherr - TestNG is internally creating multiple instances of parent module. Without the hashcode implementation, Guice ends up adding all of the parent modules into its internal map. This will need a closer/deeper look at the way in which we are creating modules and clean up such that we dont end up creating multiple modules. I think we should take it as a separate PR and not club that into this PR. So this hashcode and equals implementation cannot be removed for now.

Reply:
Ok, just create a new issue then :)

Reply:
@juherr I raised https://github.com/cbeust/testng/issues/2355 to track this.